addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/handler/VeraSceneHandler.java

```diff
@@ -0,0 +1,380 @@
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.vera2.VeraBindingConstants;
+import org.openhab.binding.vera2.config.VeraSceneConfiguration;
+import org.openhab.binding.vera2.controller.json.Scene;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VeraSceneHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dmitriy Ponomarev
+ */
+public class VeraSceneHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private ScenePolling scenePolling;
+    private ScheduledFuture<?> pollingJob;
+    private VeraSceneConfiguration mConfig = null;
+
+    public VeraSceneHandler(Thing thing) {
+        super(thing);
+        scenePolling = new ScenePolling();
+    }
+
+    private class Initializer implements Runnable {
+        @Override
+        public void run() {
+            try {
+                VeraBridgeHandler veraBridgeHandler = getVeraBridgeHandler();
+                if (veraBridgeHandler != null && veraBridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                    ThingStatusInfo statusInfo = veraBridgeHandler.getThing().getStatusInfo();
+                    logger.debug("Change scene status to bridge status: {}", statusInfo);
+
+                    // Set thing status to bridge status
+                    updateStatus(statusInfo.getStatus(), statusInfo.getStatusDetail(), statusInfo.getDescription());
+
+                    try {
+                        logger.debug("Add channels");
+                        Scene scene = veraBridgeHandler.getController().getScene(mConfig.getSceneId());
+                        if (scene != null) {
+                            logger.debug("Finded {} scene", scene.name);
+                            addSceneAsChannel(scene);
+                        }
+                    } catch (Exception e) {
+                        logger.error("{}", e.getMessage());
+                        if (getThing().getStatus() == ThingStatus.ONLINE) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                                    "Error occurred when adding scene as channel: " + mConfig.getSceneId());
+                        }
+                    }
+
+                    // Initialize scene polling
+                    if (pollingJob == null || pollingJob.isCancelled()) {
+                        logger.debug("Starting polling job at intervall {}",
+                                veraBridgeHandler.getVeraBridgeConfiguration().getPollingInterval());
+                        pollingJob = scheduler.scheduleAtFixedRate(scenePolling, 10,

```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)